src/Pos/Lrc/Worker.hs

```diff
@@ -56,17 +57,17 @@ import           Pos.WorkMode.Class         (WorkMode)
 lrcOnNewSlotWorker
     :: (WorkMode ssc m, SscWorkersClass ssc, MonadGStateCore m)
     => (WorkerSpec m, OutSpecs)
-lrcOnNewSlotWorker = localOnNewSlotWorker True $ \SlotId {..} ->
+lrcOnNewSlotWorker = recoveryCommGuard $ localOnNewSlotWorker True $ \SlotId {..} ->
     when (getSlotIndex siSlot < slotSecurityParam) $
-    (lrcSingleShot siEpoch `catch` reportError) `catch` onLrcError
+        lrcSingleShot siEpoch `catch` onLrcError
   where
-    reportError (SomeException e) = do
-        reportMisbehaviourMasked version $ "Lrc worker failed with error: " <> show e
-        throwM e
-    onLrcError UnknownBlocksForLrc =
+    onLrcError e@UnknownBlocksForLrc = do
+        reportError e
         logInfo
```

Comment:
I still have this comment http://i.imgur.com/i7cLxye.png in my mailbox. Perhaps it's a good time to change it to `logWarning`?